src/generator/connection.ts

```diff
@@ -192,11 +192,23 @@ function generateContent(session: Session<CodeModel>, imports: ImportManager): s
         if (hostParam.clientDefaultValue) {
           pointer = '*';
           text += `\tif ${hostParam.language.go!.name} == nil {\n`;
-          text += `\t\tdefaultValue := "${hostParam.clientDefaultValue}"\n`;
+          text += `\t\tdefaultValue := ${getClientDefaultValue(hostParam)}\n`;
           text += `\t\t${hostParam.language.go!.name} = &defaultValue\n`;
           text += '\t}\n';
         }
-        text += `\thostURL = strings.ReplaceAll(hostURL, "{${hostParam.language.go!.serializedName}}", ${pointer}${hostParam.language.go!.name})\n`;
+        text += `\thostURL = strings.ReplaceAll(hostURL, "{${hostParam.language.go!.serializedName}}", `;
+        switch (hostParam.schema.type) {
+          case SchemaType.Choice:
+          case SchemaType.SealedChoice:
+            text += `string(${pointer}${hostParam.language.go!.name}))\n`;
+            break;
+          case SchemaType.String:
+            text += `${pointer}${hostParam.language.go!.name})\n`;
+            break;
+          default:
```

Comment:
Do we know we handle other values here correctly, or should we die/error if we hit a case we don't know about?

Reply:
Given that this is for constructing the host URL, it seems highly unlikely that we'd ever hit this clause (I had debated making this clause throw an error but decided the code generator shouldn't be so "opinionated" here).  However, if it was something else like a number it should work.